src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -3504,6 +3573,39 @@ internal static bool IsTypeParameterizedProperty(Type t)
             return PSMethod.Create(methods[0].method.Name, this, obj, methods, isSpecial, isHidden) as T;
         }
 
+        private protected T GetDotNetMethod<T>(object obj, MemberNamePredicate predicate) where T : PSMemberInfo
+        {
+            if (!typeof(T).IsAssignableFrom(typeof(PSMethod)))
+            {
+                return null;
+            }
+
+            CacheTable typeTable = _isStatic
+                ? GetStaticMethodReflectionTable((Type)obj)
+                : GetInstanceMethodReflectionTable(obj.GetType());
+
+            var methodCacheEntry = (MethodCacheEntry)typeTable.GetOrDefault(predicate);
+            if (methodCacheEntry == null)
+            {
+                return null;
+            }
+
+            MethodBase firstMethod = methodCacheEntry[0].method;
+            var isCtor = firstMethod is ConstructorInfo;
```

Comment:
Seems like this one could just be evaluated inline in the below expression, I don't see a particular need to have this be a separate variable.